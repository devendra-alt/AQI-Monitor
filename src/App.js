import React, { useEffect } from 'react';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { fetchCitys } from './redux/features/citys';
import Home from './components/Home';
import CityDetails from './components/CityDetails';

const router = createBrowserRouter([
  {
    path: '/',
    element: <Home />,
  },
  {
    path: '/city',
    element: <CityDetails />,
  },
]);

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCitys());
  }, [dispatch]);
  return <RouterProvider router={router} />;
}

export default App;
