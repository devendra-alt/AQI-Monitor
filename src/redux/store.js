import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './features/countries';
import citysReducer from './features/citys';
import aqiReducer from './features/aqi';

export default configureStore({
  reducer: {
    countries: countryReducer,
    citys: citysReducer,
    aqi: aqiReducer,
  },
});
