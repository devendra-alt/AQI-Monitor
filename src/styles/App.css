:root {
  --font-color: #fff;
  --pink-light: #e94988;
  --pink-dark: #df4782;
  --pitch-dark-pin: #b13967;
}

.main-header {
  background: var(--pink-dark);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
}

.btn {
  outline: none;
  border: none;
  background: none;
  cursor: pointer;
  color: var(--font-color);
}

.main-title {
  font-weight: 300;
}

.seach-city-i {
  width: 100%;
  outline: none;
  border: none;
  background: var(--pitch-dark-pin);
  padding: 1rem;
  font-size: 1.4rem;
  color: var(--font-color);
  font-weight: 400;
}

::placeholder {
  color: var(--font-color);
}

.city-list {
  background: var(--pink-light);
  display: grid;
}

.city-list-item {
  padding: 1rem;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  gap: 2rem;
  min-height: 12.5rem;
  position: relative;
}

.city-list-item:nth-child(even) {
  background-color: var(--pink-dark);
}

p {
  font-size: 1.5rem;
  font-weight: 500;
}

span {
  padding: 0;
  font-size: 1rem;
  font-weight: 600;
}

.see-more {
  position: absolute;
  display: flex;
  align-items: center;
  gap: 0.5rem;
  right: 1rem;
  bottom: 1rem;
  font-size: 1.2rem;
  font-weight: 400;
  text-decoration: none;
  color: var(--font-color);
}

.see-more > * {
  display: inline;
}

.no-city-found-msg {
  text-align: center;
  text-transform: uppercase;
  font-size: 1.5rem;
  padding: 5rem;
}

.ball-triangle-loading {
  margin: 5rem auto;
  display: grid;
  place-items: center;
  justify-content: center;
}

.city-Details {
  padding: 1rem;
}

.general-city-info-data {
  text-align: right;
  background-image: url(/src/assets/earth.svg);
  background-size: contain;
  background-repeat: no-repeat;
}

.general-city-info-data > p {
  font-size: 2rem;
  font-weight: 700;
  line-height: 0.7;
  padding-bottom: 2rem;
}

.general-city-info-data > p > span::before {
  content: '\a';
  white-space: pre;
}

.aqi-data-table {
  margin: 0 auto;
  width: 100%;
}

.aqi-data-title {
  font-size: 1.3rem;
  font-weight: 500;
  text-transform: uppercase;
  background-color: var(--pink-dark);
  text-align: left;
  width: 100%;
  margin: 0;
  padding: 0;
}

.aqi-data-table > tbody > tr > td {
  text-align: left;
  font-size: 1.5rem;
  width: 50%;
}

.aqi-data-table > tbody > tr:nth-child(even) {
  background-color: var(--pink-dark);
}

.general-city-aqi-point > img {
  display: none;
}

@media screen and (min-width: 764px) {
  .city-list {
    grid-template-columns: repeat(5, 1fr);
  }

  .city-list-item {
    grid-template-rows: repeat(3, 1fr);
    grid-template-columns: 1fr;
  }

  .general-city-info {
    width: 60%;
    margin: 0 auto;
  }

  .general-city-info-data {
    background-position: 100%;
  }

  .general-city-info-data > p {
    font-size: 2.5rem;
    text-align: left;
  }

  .aqi-data-title {
    font-size: 1.5rem;
  }

  .aqi-data-table > tbody > tr > td {
    padding: 0.1rem;
    font-size: 1.5rem;
  }

  .general-city-aqi-point > img {
    display: inline;
    margin-left: 0.5rem;
  }
}
