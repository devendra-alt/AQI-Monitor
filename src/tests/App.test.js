import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/store';

describe('it should render App', () => {
  it('should render App', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
