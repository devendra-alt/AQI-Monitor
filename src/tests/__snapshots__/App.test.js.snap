// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render App should render App 1`] = `
Array [
  <header
    className="main-header"
  >
    <div
      className="main-left-back-area"
    />
    <h1
      className="main-title"
    >
      AQI monitor
    </h1>
    <div
      className="main-utils"
    >
      <button
        className="settings btn"
        type="button"
      >
        <img
          alt="settings icon"
          className="setting-img"
          height="20px"
          src="settings.svg"
          width="20px"
        />
      </button>
    </div>
  </header>,
  <section
    className="citys-state-section"
  >
    <article
      className="ball-triangle-loading"
    >
      <div
        aria-busy={true}
        aria-label="ball-triangle-loading"
        data-testid="ball-triangle-loading"
        role="status"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <svg
          data-testid="ball-triangle-svg"
          height={100}
          stroke="#fff"
          viewBox="0 0 57 57"
          width={100}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              strokeWidth="2"
              transform="translate(1 1)"
            >
              <circle
                cx="5"
                cy="50"
                r={5}
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="50;5;50;50"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="5;27;49;5"
                />
              </circle>
              <circle
                cx="27"
                cy="5"
                r={5}
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="5"
                  repeatCount="indefinite"
                  to="5"
                  values="5;50;50;5"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="27"
                  repeatCount="indefinite"
                  to="27"
                  values="27;49;5;27"
                />
              </circle>
              <circle
                cx="49"
                cy="50"
                r={5}
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="50;50;5;50"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="49"
                  repeatCount="indefinite"
                  to="49"
                  values="49;5;27;49"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </article>
  </section>,
]
`;
